File: reservation-impl/src/main/java/com/example/reservation/impl/Module.java
package com.example.reservation.impl;

import com.example.reservation.api.ReservationService;
import com.example.search.api.SearchService;
import com.google.inject.AbstractModule;
import com.lightbend.lagom.javadsl.server.ServiceGuiceSupport;

public class Module extends AbstractModule implements ServiceGuiceSupport {

  @Override
  protected void configure() {
    bindService(ReservationService.class, ReservationServiceImpl.class);
    bindClient(SearchService.class);
  }
}

File: search-impl/src/main/scala/com/example/search/impl/Module.scala
package com.example.search.impl

import com.example.reservation.api.ReservationService
import com.example.search.api.SearchService
import com.google.inject.AbstractModule
import com.lightbend.lagom.javadsl.server.ServiceGuiceSupport

class Module extends AbstractModule with ServiceGuiceSupport{
  override def configure(): Unit = {
    bindService(classOf[SearchService], classOf[SearchServiceImpl])
    bindClient(classOf[ReservationService])
  }
}
